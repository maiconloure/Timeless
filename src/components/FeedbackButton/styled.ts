import styled from 'styled-components';

export const FeedBackContainer = styled.div<{ showMobileMenu: boolean }>`
  z-index: 99;
  ${(props) =>
    !props.showMobileMenu &&
    `
      position: absolute;
      top: 58px;
      left: 10px;
  `}
  transition: 100ms;

  margin: 10px 0;

  .tooltip {
    &:hover .tooltiptext {
      visibility: visible;
    }
  }

  .tooltiptext {
    font: 500 1.5rem Inter;
    visibility: hidden;
    width: 100px;
    background-color: black;
    color: #fff;
    text-align: center;
    border-radius: 3px;
    padding: 4px 8px;

    /* Position the tooltip */
    position: absolute;
    z-index: 99999;
  }
`;

export const ConfirmFeedBack = styled.div`
  position: absolute;
  top: 50px;
  left: 55px;
  z-index: 99;
  background-color: #0190f5;
  width: 260px;
  height: 100px;
  padding: 8px 4px;
  display: flex;
  flex-direction: column;
  box-shadow: 0 0 0 4px #4169e1;
  button {
    margin: 10px 5px;
  }
`;

export const Message = styled.p`
  font: 600 1.6rem Inter;
  color: #fff;
  margin: 0 4px;
`;

export const Buttons = styled.div``;
