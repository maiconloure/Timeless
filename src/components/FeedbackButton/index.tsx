import { Button } from 'capstone-project';
import React, { useState } from 'react';

import * as St from './styled';

const FeedbackButton = () => {
  const [handleConfirm, setHandleConfirm] = useState(false);
  return (
    <St.FeedBackContainer>
      <div className="tooltip">
        <Button
          fontSize="1.9rem"
          width="140px"
          height="40px"
          radius="2px"
          onClick={() => {
            setHandleConfirm(!handleConfirm);
          }}>
          feedback
        </Button>
        <span className="tooltiptext">Ajude-nos a melhorar!</span>
      </div>

      {handleConfirm && (
        <St.ConfirmFeedBack>
          <St.Message>Você será redirecionado para um formulário do Google.</St.Message>
          <St.Buttons>
            <a
              href="https://docs.google.com/forms/d/e/1FAIpQLSeQ19Ikk1f2XLbEhYMPkdIcAtZK8T4XWC2u-gCSPJ4InUvpSA/viewform"
              target="_blank"
              rel="noopener noreferrer">
              <Button
                fontSize="1.8rem"
                width="100px"
                height="30px"
                radius="2px"
                onClick={() => {
                  setHandleConfirm(false);
                }}>
                permitir
              </Button>
            </a>
            <Button
              fontSize="1.8rem"
              width="90px"
              height="30px"
              radius="2px"
              onClick={() => {
                setHandleConfirm(false);
              }}>
              negar
            </Button>
          </St.Buttons>
        </St.ConfirmFeedBack>
      )}
    </St.FeedBackContainer>
  );
};

export default FeedbackButton;
