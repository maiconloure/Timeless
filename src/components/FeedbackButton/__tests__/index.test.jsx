import React, { useState } from 'react';
import FeedbackButton from '../index';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(() => jest.fn()),
}));

describe('Snapshot', () => {
  it('renders', () => {
    const setState = jest.fn();
    useState.mockImplementation(() => [false, setState]);
    const tree = renderer.create(<FeedbackButton />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders opened', () => {
    const setState = jest.fn();
    useState.mockImplementation(() => [true, setState]);
    const tree = renderer.create(<FeedbackButton />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Behavior', () => {
  it('opens the feedback options', () => {
    const setState = jest.fn();
    useState.mockImplementation((init) => [init, setState]);
    const wrapper = mount(<FeedbackButton />);

    wrapper.find('button').at(0).simulate('click');
    expect(setState).toHaveBeenCalledTimes(1);
  });
});
