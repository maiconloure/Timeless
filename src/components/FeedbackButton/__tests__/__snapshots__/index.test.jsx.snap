// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div
  className="sc-efHXLn dFKHlR"
>
  <div
    className="tooltip"
  >
    <button
      className="sc-bdnylx gynFlg"
      color="#fff"
      fontSize="1.9rem"
      height="40px"
      onClick={[Function]}
      radius="2px"
      width="140px"
    >
      feedback
    </button>
    <span
      className="tooltiptext"
    >
      Ajude-nos a melhorar!
    </span>
  </div>
</div>
`;

exports[`Snapshot renders opened 1`] = `
<div
  className="sc-efHXLn dFKHlR"
>
  <div
    className="tooltip"
  >
    <button
      className="sc-bdnylx gynFlg"
      color="#fff"
      fontSize="1.9rem"
      height="40px"
      onClick={[Function]}
      radius="2px"
      width="140px"
    >
      feedback
    </button>
    <span
      className="tooltiptext"
    >
      Ajude-nos a melhorar!
    </span>
  </div>
  <div
    className="sc-cTJmaU cxHdbX"
  >
    <p
      className="sc-jNnnWF jsfpUR"
    >
      Você será redirecionado para um formulário do Google.
    </p>
    <div
      className="sc-dPaNSN"
    >
      <a
        href="https://docs.google.com/forms/d/e/1FAIpQLSeQ19Ikk1f2XLbEhYMPkdIcAtZK8T4XWC2u-gCSPJ4InUvpSA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          className="sc-bdnylx dWattX"
          color="#014D82"
          fontSize="1.8rem"
          height="30px"
          onClick={[Function]}
          radius="2px"
          width="100px"
        >
          permitir
        </button>
      </a>
      <button
        className="sc-bdnylx kBElPH"
        color="#014D82"
        fontSize="1.8rem"
        height="30px"
        onClick={[Function]}
        radius="2px"
        width="90px"
      >
        negar
      </button>
    </div>
  </div>
</div>
`;
